src/runtime/lpython_builtin.py

```diff
@@ -100,6 +100,122 @@ def abs(c: c64) -> f64:
     b = _lfortran_zaimag(c)
     return (a**2.0 + b**2.0)**(1/2)
 
+# all
+# supported data types: i32, i64, f32, f64, bool
+
+@overload
+def all(arr: list[i32]) -> bool:
+    """
+    Return True if all elements of the iterable are true (or if the iterable is empty).
+    """
+    i: i32
+    for i in range(len(arr)):
+        if not arr[i]:
+            return False
+    return True
+
+@overload
+def all(arr: list[i64]) -> bool:
+    """
+    Return True if all elements of the iterable are true (or if the iterable is empty).
+    """
+    i: i32
+    for i in range(len(arr)):
+        if not arr[i]:
+            return False
+    return True
+
+@overload
+def all(arr: list[f32]) -> bool:
+    """
+    Return True if all elements of the iterable are true (or if the iterable is empty).
+    """
+    i: i32
+    for i in range(len(arr)):
+        if not arr[i]:
+            return False
+    return True
+
+@overload
+def all(arr: list[f64]) -> bool:
+    """
+    Return True if all elements of the iterable are true (or if the iterable is empty).
+    """
+    i: i32
+    for i in range(len(arr)):
+        if not arr[i]:
+            return False
+    return True
+
+@overload
+def all(arr: list[bool]) -> bool:
+    """
+    Return True if all elements of the iterable are true (or if the iterable is empty).
+    """
+    i: i32
+    for i in range(len(arr)):
+        if not arr[i]:
+            return False
+    return True
+
+# any
+# supported data types: i32, i64, f32, f64, bool
+
+@overload
+def any(arr: list[i32]) -> bool:
+    """
+    Return True if any element of the iterable is true. If the iterable is empty, return False.
+    """
+    i: i32
+    for i in range(len(arr)):
+        if arr[i]:
+            return True
+    return False
+
+@overload
+def any(arr: list[i64]) -> bool:
+    """
+    Return True if any element of the iterable is true. If the iterable is empty, return False.
+    """
+    i: i32
+    for i in range(len(arr)):
+        if arr[i]:
+            return True
+    return False
+
+@overload
+def any(arr: list[f32]) -> bool:
+    """
+    Return True if any element of the iterable is true. If the iterable is empty, return False.
+    """
+    i: i32
+    for i in range(len(arr)):
+        if arr[i]:
+            return True
+    return False
+
+@overload
+def any(arr: list[f64]) -> bool:
+    """
+    Return True if any element of the iterable is true. If the iterable is empty, return False.
+    """
+    i: i32
+    for i in range(len(arr)):
+        if arr[i]:
+            return True
+    return False
+
+@overload
+def any(arr: list[bool]) -> bool:
+    """
+    Return True if any element of the iterable is true. If the iterable is empty, return False.
+    """
+    i: i32
+    for i in range(len(arr)):
+        if arr[i]:
+            return True
+    return False
```

Comment:
I feel generics usage can prevent a lot of code repetition here. @certik Can we use generics for these kind of purposes? The other way is to generate these functions on demand using ASR APIs (that's the most optimal) because `all`, `any` can take any iterable as input (list, set, tuple, etc.).

Reply:
I tried using generics. However, I think this will only work for lists but I am not sure.

Reply:
Its cool. My above comment is for long term approach. If @certik agrees then we will have to add a `TODO` for all the implementations here so that we don't forget to replace them with a better one.